usage message e2e failure after #5216

Unfortunately, full e2e did not run on #5216 so this has come up after merge:

```
 --- FAIL: TestE2E/PAR/BUILD/build_encrypted_with_passphrase/passphrase_on_cmdline (0.12s)
                    imgbuild.go:979: Running command "/usr/local/bin/singularity build --encrypt --passphrase e2e-passphrase /tmp/stest.199202312/passphrase-encryption.146989616/dummy_encrypted_env_var.sif library://alpine:latest"
                    imgbuild.go:979: failed to match pattern: Command "/usr/local/bin/singularity build --encrypt --passphrase e2e-passphrase /tmp/stest.199202312/passphrase-encryption.146989616/dummy_encrypted_env_var.sif library://alpine:latest":
                        Expect error stream match regular expression:
                        ^Error for command \"build\": accepts 2 arg\(s\), received 3
                        Command error output:
                        Error for command "singularity": accepts 2 arg(s), received 3
                        
                        Usage:
                          singularity [global options...] <command>
                        
                        Available Commands:
                          build       Build a Singularity image
                          cache       Manage the local cache
                          capability  Manage Linux capabilities for users and groups
                          config      Manage various singularity configuration (root user only)
                          delete      Deletes requested image from the library
```

--- a/cmd/internal/cli/singularity.go
+++ b/cmd/internal/cli/singularity.go
@@ -450,17 +450,26 @@ func ExecuteSingularity() {
 	}()
 
 	if err := singularityCmd.ExecuteContext(ctx); err != nil {
-		name := singularityCmd.Name()
+		// Find the subcommand to display more useful help, and the correct
+		// subcommand name in messages - i.e. 'run' not 'singularity'
+		// This is required because we previously used ExecuteC that returns the
+		// subcommand... but there is no ExecuteC that variant accepts a context.
+		subCmd, _, subCmdErr := singularityCmd.Find(args[1:])
+		if subCmdErr != nil {
+			singularityCmd.Printf("Error: %v\n\n", subCmdErr)
+		}
+
+		name := subCmd.Name()
 		switch err.(type) {
 		case cmdline.FlagError:
-			usage := singularityCmd.Flags().FlagUsagesWrapped(getColumns())
+			usage := subCmd.Flags().FlagUsagesWrapped(getColumns())
 			singularityCmd.Printf("Error for command %q: %s\n\n", name, err)
 			singularityCmd.Printf("Options for %s command:\n\n%s\n", name, usage)
 		case cmdline.CommandError:
-			singularityCmd.Println(singularityCmd.UsageString())
+			singularityCmd.Println(subCmd.UsageString())
 		default:
 			singularityCmd.Printf("Error for command %q: %s\n\n", name, err)
-			singularityCmd.Println(singularityCmd.UsageString())
+			singularityCmd.Println(subCmd.UsageString())
 		}
 		singularityCmd.Printf("Run '%s --help' for more detailed usage information.\n",
 			singularityCmd.CommandPath())

